Src/Particle/AMReX_ParticleIO.H

```diff
@@ -588,11 +588,7 @@ ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>
 
         // Only write out valid particles.
         int cnt = 0;
-        for (int k = 0; k < kv.second.GetArrayOfStructs().numParticles(); ++k)
-        {
-            if (pflags[k]) cnt++;
-        }
-
+        particle_detail::countFlags(pflags, cnt);
```

Comment:
Minor comment:  Would it be better to do `int cnt = particle_detail::countFlags(pflags)` instead?

Reply:
Yes, but I was using the return type for `enable_if` for these functions so I stuck with that. I could change it if you have a strong opinion, but there aren't user-facing functions.

Reply:
std::enable_if_t has an optional parameter.

Reply:
oh, cool.